drivers/reset/reset.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2022 Andrei-Edward Popa
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT reset
```

Comment:
do you have any hardware implementing this? This looks like a reset controller leveraging some syscon area.

Reply:
> do you have any hardware implementing this? This looks like a reset controller leveraging some syscon area.

Yes, I have an implementation for Raspberry Pi Pico, I will update the code soon.